frame/staking/src/lib.rs

```diff
@@ -2191,6 +2849,134 @@ impl<T, Reporter, Offender, R, O> ReportOffence<Reporter, Offender, O>
 	}
 }
 
+/// Disallows any transactions that change the election result to be submitted after the election
+/// window is open.
+#[derive(Encode, Decode, Clone, Eq, PartialEq)]
+pub struct LockStakingStatus<T>(sp_std::marker::PhantomData<T>);
+
+impl<T: Trait + Send + Sync> sp_std::fmt::Debug for LockStakingStatus<T> {
+	fn fmt(&self, f: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
+		write!(f, "LockStakingStatus")
+	}
+}
+
+impl<T> LockStakingStatus<T> {
+	/// Create new `LockStakingStatus`.
+	pub fn new() -> Self {
+		Self(sp_std::marker::PhantomData)
+	}
+}
+
+impl<T> Default for LockStakingStatus<T> {
+	fn default() -> Self {
+		Self::new()
+	}
+}
+
+impl<T: Trait + Send + Sync> SignedExtension for LockStakingStatus<T> {
+	const IDENTIFIER: &'static str = "LockStakingStatus";
+	type AccountId = T::AccountId;
+	type Call = <T as Trait>::Call;
+	type AdditionalSigned = ();
+	type DispatchInfo = frame_support::weights::DispatchInfo;
+	type Pre = ();
+
+	fn additional_signed(&self) -> Result<(), TransactionValidityError> { Ok(()) }
+
+	fn validate(
+		&self,
+		_who: &Self::AccountId,
+		call: &Self::Call,
+		_info: Self::DispatchInfo,
+		_len: usize,
+	) -> TransactionValidity {
+		if let Some(inner_call) = call.is_sub_type() {
+			if let ElectionStatus::Open(_) = <Module<T>>::era_election_status() {
+				match inner_call {
+					Call::<T>::set_payee(..) |
+					Call::<T>::set_controller(..) |
+					Call::<T>::set_validator_count(..) |
+					Call::<T>::force_no_eras(..) |
+					Call::<T>::force_new_era(..) |
+					Call::<T>::set_invulnerables(..) |
+					Call::<T>::force_unstake(..) |
+					Call::<T>::force_new_era_always(..) |
+					Call::<T>::cancel_deferred_slash(..) |
+					Call::<T>::set_history_depth(..) |
+					Call::<T>::reap_stash(..) |
+					Call::<T>::submit_election_solution(..) |
+					Call::<T>::submit_election_solution_unsigned(..) => {
+						// These calls are allowed. Nothing.
+					}
+					_ => {
+						return Err(InvalidTransaction::Stale.into());
+					}
+				}
+			}
+		}
+
+		Ok(Default::default())
+	}
+}
+
+#[allow(deprecated)]
+impl<T: Trait> frame_support::unsigned::ValidateUnsigned for Module<T> {
+	type Call = Call<T>;
+	fn validate_unsigned(call: &Self::Call) -> TransactionValidity {
+		if let Call::submit_election_solution_unsigned(
+			_,
+			_,
+			score,
+		) = call {
+			use offchain_election::DEFAULT_LONGEVITY;
+
+			// discard solution not coming from the local OCW.
+			// TODO: ...
```

Comment:
Once we get this via https://github.com/paritytech/substrate/pull/5366 it should be all good to go. At the current state everything also works but this is basically a DOS verctor to allow any unsigned transaction to be validated.